lib/logstash/gemfile.rb

```diff
@@ -0,0 +1,173 @@
+module LogStash
+
+  class GemfileError < StandardError; end
+
+  class Gemfile
+    attr_accessor :gemset
+
+    HEADER = \
+      "# This is a Logstash generated Gemfile.\n" + \
+      "# If you modify this file manually all comments and formatting will be lost.\n\n"
+
+    # @params io [IO] any IO object that supports read, write, truncate, rewind
+    def initialize(io)
+      @io = io
+      @gemset = nil
+    end
+
+    def load
+      @gemset ||= Dsl.parse(@io.read)
```

Comment:
This is obviously up for style battling, but my thoughts are:
* Constants are required to start with a capital letter, but have no other requirements.
* Dsl is not a word, like Socket, it's an acronym :)
* DSL is an acronym that happens to already have a capital D to start.

^^ Logic for my "DSL". 

Nobody seems to be consistent on this. Ruby core/stdlib maybe seems to use my style?

```
>> Module.constants.grep(/^[A-Z]{2,}[a-z]/)
=> [:IOError, :EOFError, :IPSocket, :TCPSocket, :TCPServer, :UDPSocket, :UNIXSocket, :UNIXServer]
```

CamelCase for words, but acronyms stay CAPITALIZED ?